.github/workflows/k8s-pr-test.yml

```diff
@@ -0,0 +1,78 @@
+# About:
+# This action runs when a non-doc Pull Request has been made to the
+# stack directory or any of it subdirectories. This action will test
+# in the same manner the Addon Service applies K8s 1-clicks,
+# using the deploy.sh script and relaying on rollout status to
+# confirm a successfull deploy. This action does NOT test
+# functionality in the k8 1-click app.
+
+# Note:
+# A DigitalOcean token needs to be saved in the repo 
+# Settings --> Secrets with the name, "DIGITALOCEAN_ACCESS_TOKEN"
+# DigitalOcean Tokens doc: 
+#   https://www.digitalocean.com/docs/apis-clis/api/create-personal-access-token/
+# 
+# A 2vCPU 2GB Memory 3 Node cluster in SFO2 is spun up to test. The cluster
+# name includes the Pull Request number. 
+# Once the rollout finishes the cluster is deleted.
+# 
+# This action also makes use of the DOCTL Action:
+#   https://github.com/digitalocean/action-doctl
+
+name: DigitalOcean Kubernetes Test
+on:
+  pull_request:
+    # Do not test if the only update was a doc update,
+    # ie new or edit of an existing markdown file.
+    # Only test if a change was made to the stacks subdirectories.
+    # This assumes any change to source would have to be rendered.
+    paths:
+    - 'stacks/**'    
```

Comment:
This would only run integrations when there are new apps added, not when yaml config changes for existing apps.

You'd also want to test if there were changes to the src directory. The src directory is where we put the helm charts and yaml for each stack.

The stacks directory defines the 1-Click app as one or more charts from the src directory.

Reply:
Correct, and this is intentional. The idea of this testing is to reflect and give feedback on how DigitalOcean applies 1-clicks. Thus, until the stacks directory is updated the 1-click will not run any updates made to the src directory. Our Contributor docs do ask users to run render when the updates are made to src.